Sources/AppCoinsSDK/Models/Services/Gamification/AppCoinGamificationServiceClient.swift

```diff
@@ -27,14 +27,21 @@ internal class AppCoinGamificationServiceClient : AppCoinGamificationService {
             let task = URLSession.shared.dataTask(with: request) { data, response, error in
                 if let error = error {
                     if let nsError = error as NSError?, nsError.code == NSURLErrorNotConnectedToInternet {
-                        result(.failure(.noInternet))
+                        result(.failure(.noInternet(message: "Internet Connection Failed", description: "Could not get internet connection to \(url)")))
                     } else {
-                        result(.failure(.failed()))
+                        result(.failure(.failed(message: "Service Failed", description: "Failed to communicate with service on endpoint: \(url)")))
                     }
                 } else {
-                    if let data = data, let findResult = try? JSONDecoder().decode(TransactionBonusRaw.self, from: data) {
-                        result(.success(findResult))
-                    } else { result(.failure(.failed())) }
+                    do {
+                        if let data = data {
+                            let successResponse = try JSONDecoder().decode(TransactionBonusRaw.self, from: data)
+                            result(.success(successResponse))
+                        } else {
+                            result(.failure(.failed(message: "Service Failed", description: "No data received from endpoint: \(url)")))
```

Comment:
Also add code location on the description or service layer errors.

Reply:
Done